src/control/crawler.c

```diff
@@ -214,116 +216,257 @@ static void dt_control_crawler_response_callback(GtkWidget *dialog, gint respons
 {
   dt_control_crawler_gui_t *gui = (dt_control_crawler_gui_t *)user_data;
-
-  gboolean selected = gtk_toggle_button_get_active(togglebutton);
+  GtkTreeSelection *selection = gtk_tree_view_get_selection(gui->tree);
 
   GtkTreeIter iter;
   gboolean valid = gtk_tree_model_get_iter_first(gui->model, &iter);
   while(valid)
   {
-    gtk_list_store_set(GTK_LIST_STORE(gui->model), &iter, DT_CONTROL_CRAWLER_COL_SELECTED, selected, -1);
+    if(gtk_tree_selection_iter_is_selected(selection, &iter))
+      gtk_tree_selection_unselect_iter(selection, &iter);
+    else
+      gtk_tree_selection_select_iter(selection, &iter);
+
     valid = gtk_tree_model_iter_next(gui->model, &iter);
   }
 }
 
-// reload xmp files of the selected images
-static void _reload_button_clicked(GtkButton *button, gpointer user_data)
+
+static void _db_update_timestamp(const int id, const time_t timestamp)
 {
-  dt_control_crawler_gui_t *gui = (dt_control_crawler_gui_t *)user_data;
+  // Update DB writing timestamp with XMP file timestamp
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "UPDATE main.images SET write_timestamp = ?2 WHERE id = ?1", -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, id);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 2, timestamp);
+  sqlite3_step(stmt);
+  sqlite3_finalize(stmt);
+}
 
-  GtkTreeIter iter;
-  gboolean valid = gtk_tree_model_get_iter_first(gui->model, &iter);
-  while(valid)
+
+static void _get_crawler_entry_from_model(GtkTreeModel *model, GtkTreeIter *iter,
+                                          dt_control_crawler_result_t *entry)
+{
+  gtk_tree_model_get(model, iter, DT_CONTROL_CRAWLER_COL_IMAGE_PATH, &entry->image_path, DT_CONTROL_CRAWLER_COL_ID,
+                     &entry->id, DT_CONTROL_CRAWLER_COL_XMP_PATH, &entry->xmp_path,
+                     DT_CONTROL_CRAWLER_COL_TS_DB_INT, &entry->timestamp_db, DT_CONTROL_CRAWLER_COL_TS_XMP_INT,
+                     &entry->timestamp_xmp, -1);
+}
+
+
+static void _append_row_to_remove(GtkTreeModel *model, GtkTreePath *path, gpointer user_data)
+{
+  // append TreeModel rows to the list to remove
+  GList **rowref_list = (GList **)user_data;
+  GtkTreeRowReference *rowref = gtk_tree_row_reference_new(model, path);
+  *rowref_list = g_list_append(*rowref_list, rowref);
+}
+
+
+static void _cleanup_GList(GList *list)
+{
+  g_list_foreach(list, (GFunc) gtk_tree_row_reference_free, NULL);
+  g_list_free(list);
+}
+
+
+static void sync_xmp_to_db(GtkTreeModel *model, GtkTreePath *path, GtkTreeIter *iter, gpointer user_data)
+{
+  dt_control_crawler_result_t entry = { 0 };
+  _get_crawler_entry_from_model(model, iter, &entry);
+  _db_update_timestamp(entry.id, entry.timestamp_xmp);
+  const int success = dt_history_load_and_apply(entry.id, entry.xmp_path, 0);  // success = 0, fail = 1
+
+  if(!success) _append_row_to_remove(model, path, user_data);
+  fprintf(stdout, "%s synced XMP -> DB\n", entry.image_path);
```

Comment:
I'm not sure we want this. Most of the time the console is not visible. And lot of files have been synchronized then the list will certainly not be fully visible. All in all, I would remove all the stdout output (more below) or create a new log file (with a date timestamp) and write those information there.

Reply:
See new commit